recipes-sota/auto-provisioning/auto-provisioning/auto-provisioning.sh

```diff
@@ -13,111 +15,122 @@ PROV_ACCESS_TOKEN=""
 
 PROV_REGISTER_ENDPOINT=${PROV_REGISTER_ENDPOINT:-"https://app.torizon.io/api/accounts/devices"}
 
-function log()
-{
+log() {
```

Comment:
Hi @rborn-tx I know this is not new, but since I noticed it now I wanted to point it out here.
There is a built-in shell command called “log” that may clash with this function in the future if this command is available in the system.

That said, I don't think we'll ever purposely add this command into our system because we don't need it, but I think it's good to keep in mind that if someday, after updating our shell, we experience strange problems, this could be one of the reasons.

Usually, I use always and underscore before the function name when I'm in doubt, just to be sure that in the future we will avoid all possible clashes.

Reply:
@EstebanSannin `log` is not a built-in on Bash and we're explicitly using it as shell here. Anyways, with Bash, even if there was a built-in named `log`, the function with the same name would take precedence. In fact, Bash has commands [builtin](https://www.gnu.org/software/bash/manual/bash.html#index-builtin) and [command](https://www.gnu.org/software/bash/manual/bash.html#index-command) to allow one to explicitly call the built-in or external command respectively in case they have been overridden. 

Thus, I wouldn't expect any issues now or in the future. But, thanks for pointing out your concern :smiley_cat:.